internal/daemon/controller/handlers/authmethods/authmethod_service_test.go

```diff
@@ -1550,3 +1599,491 @@ func TestCreate(t *testing.T) {
+	req.Filter = fmt.Sprintf(`"/item/id"==%q or "/item/id"==%q`, allAuthMethods[len(allAuthMethods)-2].Id, allAuthMethods[len(allAuthMethods)-1].Id)
+	got, err = s.ListAuthMethods(ctx, req)
+	require.NoError(t, err)
+	require.Len(t, got.GetItems(), 1)
+	assert.Empty(
+		t,
+		cmp.Diff(
+			got,
+			&pbs.ListAuthMethodsResponse{
+				Items:        []*pb.AuthMethod{allAuthMethods[len(allAuthMethods)-2]},
+				ResponseType: "delta",
+				ListToken:    "",
+				SortBy:       "created_time",
+				SortDir:      "desc",
+				// Should be empty again
+				RemovedIds:   nil,
+				EstItemCount: 10,
+			},
+			cmpopts.SortSlices(func(a, b string) bool {
+				return a < b
+			}),
+			cmpopts.SortSlices(func(a, b protocmp.Message) bool {
+				return a.String() < b.String()
+			}),
+			protocmp.Transform(),
+			protocmp.IgnoreFields(&pb.OidcAuthMethodAttributes{}, "client_secret_hmac"),
+			protocmp.IgnoreFields(&pb.LdapAuthMethodAttributes{}, "bind_password_hmac", "client_certificate_key_hmac"),
+			protocmp.IgnoreFields(&pbs.ListAuthMethodsResponse{}, "list_token"),
+		),
+	)
+	req.ListToken = got.ListToken
+	// Get the second page
+	got, err = s.ListAuthMethods(ctx, req)
+	require.NoError(t, err)
+	require.Len(t, got.GetItems(), 1)
+	assert.Empty(
+		t,
+		cmp.Diff(
+			got,
+			&pbs.ListAuthMethodsResponse{
+				Items:        []*pb.AuthMethod{allAuthMethods[len(allAuthMethods)-1]},
+				ResponseType: "complete",
+				ListToken:    "",
+				SortBy:       "created_time",
+				SortDir:      "desc",
+				RemovedIds:   nil,
+				EstItemCount: 10,
+			},
+			cmpopts.SortSlices(func(a, b string) bool {
+				return a < b
+			}),
+			cmpopts.SortSlices(func(a, b protocmp.Message) bool {
+				return a.String() < b.String()
+			}),
+			protocmp.Transform(),
+			protocmp.IgnoreFields(&pb.OidcAuthMethodAttributes{}, "client_secret_hmac"),
+			protocmp.IgnoreFields(&pb.LdapAuthMethodAttributes{}, "bind_password_hmac", "client_certificate_key_hmac"),
+			protocmp.IgnoreFields(&pbs.ListAuthMethodsResponse{}, "list_token"),
+		),
+	)
+	req.ListToken = got.ListToken
+
+	// List items in the empty scope
+	req = &pbs.ListAuthMethodsRequest{
+		ScopeId:   orgNoAms.PublicId,
+		Filter:    "",
+		ListToken: "",
+		PageSize:  2,
+	}
+	got, err = s.ListAuthMethods(ctx, req)
+	require.NoError(t, err)
+	require.Len(t, got.GetItems(), 0)
+	// Compare without comparing the list token
+	assert.Empty(
+		t,
+		cmp.Diff(
+			got,
+			&pbs.ListAuthMethodsResponse{
+				Items:        nil,
+				ResponseType: "complete",
+				ListToken:    "",
+				SortBy:       "created_time",
+				SortDir:      "desc",
+				RemovedIds:   nil,
+			},
+			cmpopts.SortSlices(func(a, b string) bool {
+				return a < b
+			}),
+			cmpopts.SortSlices(func(a, b protocmp.Message) bool {
+				return a.String() < b.String()
+			}),
+			cmpopts.SortSlices(func(a, b string) bool {
+				return a < b
+			}),
+			cmpopts.SortSlices(func(a, b protocmp.Message) bool {
+				return a.String() < b.String()
+			}),
```

Comment:
🚨.

This got me thinking though, maybe we should just have a `cmpOpts` variable like we do in the service domain tests for these. Would make this test a bit shorter. What do you think?